okhttp-tls/src/main/java/okhttp3/tls/HandshakeCertificates.kt

```diff
@@ -67,20 +67,49 @@ import javax.net.ssl.X509TrustManager
  *    roots private to an organization or service.
  */
 class HandshakeCertificates private constructor(
-  private val keyManager: X509KeyManager,
-  private val trustManager: X509TrustManager
+  @get:JvmName("keyManager") val keyManager: X509KeyManager,
+  @get:JvmName("trustManager") val trustManager: X509TrustManager
 ) {
+
+  @JvmName("-deprecated_keyManager")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "keyManager"),
+      level = DeprecationLevel.WARNING)
   fun keyManager(): X509KeyManager = keyManager
 
+  @JvmName("-deprecated_trustManager")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "trustManager"),
+      level = DeprecationLevel.WARNING)
   fun trustManager(): X509TrustManager = trustManager
 
-  fun sslSocketFactory(): SSLSocketFactory = sslContext().socketFactory
+  @get:JvmName("sslSocketFactory") val sslSocketFactory: SSLSocketFactory
+    get() {
+      return sslContext.socketFactory
```

Comment:
I'd argue against this being a val, it calls functions and creates a new heavyweight object, in fact two in a chain.